pkg/deploy/controller/deployment/controller.go

```diff
@@ -0,0 +1,189 @@
+)
+
+// DeploymentController starts a deployment by creating a deployer pod which
+// implements a deployment strategy. The status of the deployment will follow
+// the status of the deployer pod.
+//
+// Use the DeploymentControllerFactory to create this controller.
+type DeploymentController struct {
+	// deploymentClient provides access to deployments.
+	deploymentClient deploymentClient
+	// podClient provides access to pods.
+	podClient podClient
+	// makeContainer knows how to make a container appropriate to execute a deployment strategy.
+	makeContainer func(strategy *deployapi.DeploymentStrategy) (*kapi.Container, error)
+	// decodeConfig knows how to decode the deploymentConfig from a deployment's annotations.
+	decodeConfig func(deployment *kapi.ReplicationController) (*deployapi.DeploymentConfig, error)
+}
+
+// fatalError is an error which can't be retried.
+type fatalError string
+
+func (e fatalError) Error() string { return "fatal error handling deployment: " + string(e) }
+
+// Handle processes a new deployment and creates a new Pod which implements the specific
+// deployment behavior. The deployment and pod are correlated with annotations. If the pod was
+// successfully created, the deployment's status is transitioned to pending.
+func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) error {
+	currentStatus := statusFor(deployment)
+	nextStatus := currentStatus
+
+	switch currentStatus {
+	case deployapi.DeploymentStatusNew:
+		podTemplate, err := c.makeDeployerPod(deployment)
+		if err != nil {
+			return fatalError(fmt.Sprintf("couldn't make deployer pod for %s: %v", labelForDeployment(deployment), err))
+		}
+
+		deploymentPod, err := c.podClient.createPod(deployment.Namespace, podTemplate)
+		if err != nil {
+			// If the pod already exists, it's possible that a previous CreatePod succeeded but
+			// the deployment state update failed and now we're re-entering.
+			if !kerrors.IsAlreadyExists(err) {
+				return fmt.Errorf("couldn't create deployer pod for %s: %v", labelForDeployment(deployment), err)
+			}
+		} else {
+			glog.V(2).Infof("Created pod %s for deployment %s", deploymentPod.Name, labelForDeployment(deployment))
+		}
+
+		deployment.Annotations[deployapi.DeploymentPodAnnotation] = deploymentPod.Name
+		nextStatus = deployapi.DeploymentStatusPending
+	case deployapi.DeploymentStatusPending,
+		deployapi.DeploymentStatusRunning,
+		deployapi.DeploymentStatusFailed:
+		glog.V(4).Infof("Ignoring deployment %s (status %s)", labelForDeployment(deployment), currentStatus)
+	case deployapi.DeploymentStatusComplete:
+		// Automatically clean up successful pods
+		// TODO: Could probably do a lookup here to skip the delete call, but it's not worth adding
+		// yet since (delete retries will only normally occur during full a re-sync).
+		podName := deployment.Annotations[deployapi.DeploymentPodAnnotation]
+		if err := c.podClient.deletePod(deployment.Namespace, podName); err != nil {
```

Comment:
I think you may have mis-read this. If the error's a NotFound error, ignore it (since the pod was already deleted). If the error is NOT a NotFound, propagate it as non-fatal (for retry).